servicetalk-http-api/src/main/java/io/servicetalk/http/api/DefaultHttpLoadBalancerFactory.java

```diff
@@ -0,0 +1,204 @@
+    }
+
+    @SuppressWarnings("deprecation")
+    @Override
+    public <T extends FilterableStreamingHttpLoadBalancedConnection> LoadBalancer<T> newLoadBalancer(
+            final String targetResource,
+            final Publisher<? extends Collection<? extends ServiceDiscovererEvent<ResolvedAddress>>> eventPublisher,
+            final ConnectionFactory<ResolvedAddress, T> connectionFactory) {
+        return rawFactory.newLoadBalancer(targetResource, eventPublisher, connectionFactory);
+    }
+
+    @Override
+    public LoadBalancer<FilterableStreamingHttpLoadBalancedConnection> newLoadBalancer(
+            final Publisher<? extends Collection<? extends ServiceDiscovererEvent<ResolvedAddress>>> eventPublisher,
+            final ConnectionFactory<ResolvedAddress, FilterableStreamingHttpLoadBalancedConnection> connectionFactory,
+            final String targetResource) {
+        return rawFactory.newLoadBalancer(eventPublisher, connectionFactory, targetResource);
+    }
+
+    @Override   // FIXME: 0.43 - remove deprecated method
+    public FilterableStreamingHttpLoadBalancedConnection toLoadBalancedConnection(
+            final FilterableStreamingHttpConnection connection) {
+        return new DefaultFilterableStreamingHttpLoadBalancedConnection(connection);
+    }
+
+    @Override
+    public FilterableStreamingHttpLoadBalancedConnection toLoadBalancedConnection(
+            final FilterableStreamingHttpConnection connection,
+            final ReservableRequestConcurrencyController concurrencyController,
+            @Nullable final ContextMap context) {
+        return new HttpLoadBalancerFactory.DefaultFilterableStreamingHttpLoadBalancedConnection(connection,
+                concurrencyController);
+    }
+
+    @Override
+    public HttpExecutionStrategy requiredOffloads() {
+        return strategy;
+    }
+
+    /**
+     * A builder for creating instances of {@link DefaultHttpLoadBalancerFactory}.
+     *
+     * @param <ResolvedAddress> The type of address after resolution for the {@link HttpLoadBalancerFactory} built by
+     * this builder.
+     */
+    public static final class Builder<ResolvedAddress> {
+        private final LoadBalancerFactory<ResolvedAddress, FilterableStreamingHttpLoadBalancedConnection> rawFactory;
+        private final HttpExecutionStrategy strategy;
+
+        private Builder(
+                final LoadBalancerFactory<ResolvedAddress, FilterableStreamingHttpLoadBalancedConnection> rawFactory,
+                final HttpExecutionStrategy strategy) {
+            this.rawFactory = rawFactory;
+            this.strategy = strategy;
+        }
+
+        /**
+         * Builds a {@link HttpLoadBalancerFactory} using the properties configured on this builder.
+         *
+         * @return A {@link HttpLoadBalancerFactory}.
+         */
+        public HttpLoadBalancerFactory<ResolvedAddress> build() {
+            return new DefaultHttpLoadBalancerFactory<>(rawFactory, strategy);
+        }
```

Comment:
Note: I widened the type to the interface. I don't know that it changes much in the future other than we can indirect in the future if we want.

Reply:
If we keep the builder (looks like we don't), it's better to use the class type bcz this is what this builder is for. We usually return an interface type when the builder is a top level class and the returned impl is an internal detail.